emmet-core/emmet/core/qchem/calculation.py

```diff
@@ -0,0 +1,669 @@
+        qcinput_kwargs
+            Additional keyword arguments that will be passed to the qcinput file
+        qcoutput_kwargs
+            Additional keyword arguments that will be passed to the qcoutput file
+
+        Returns
+        -------
+        Calculation
+            A QChem calculation document.
+        """
+
+        dir_name = Path(dir_name)
+        qcinput_file = dir_name / qcinput_file
+        qcoutput_file = dir_name / qcoutput_file
+
+        output_file_paths = _find_qchem_files(dir_name)
+
+        qcinput_kwargs = qcinput_kwargs if qcinput_kwargs else {}
+        qcinput = QCInput.from_file(qcinput_file, **qcinput_kwargs)
+        # print(qcinput)
+
+        qcoutput_kwargs = qcoutput_kwargs if qcoutput_kwargs else {}
+        qcoutput = QCOutput(qcoutput_file, **qcoutput_kwargs)
+
+        completed_at = str(datetime.fromtimestamp(qcoutput_file.stat().st_mtime))
+
+        input_doc = CalculationInput.from_qcinput(qcinput)
+        output_doc = CalculationOutput.from_qcoutput(qcoutput)
+
+        has_qchem_completed = (
+            QChemStatus.SUCCESS
+            if qcoutput.data.get("completion", [])
+            else QChemStatus.FAILED
+        )
+
+        if store_energy_trajectory:
+            print("Still have to figure the energy trajectory")
+
+        return cls(
+            dir_name=str(dir_name),
+            task_name=task_name,
+            qchem_version=qcoutput.data["version"],
+            has_qchem_completed=has_qchem_completed,
+            completed_at=completed_at,
+            input=input_doc,
+            output=output_doc,
+            output_file_paths={
+                k.lower(): Path(v)
+                if isinstance(v, str)
+                else {k2: Path(v2) for k2, v2 in v.items()}
+                for k, v in output_file_paths.items()
+            },
+            level_of_theory=level_of_theory(input_doc),
+            solvation_lot_info=lot_solvent_string(input_doc),
+            task_type=task_type(input_doc),
+            calc_type=calc_type(input_doc),
+        )
+
+
+def _find_qchem_files(
+    path: Union[str, Path],
+) -> Dict[str, Any]:
+    """
+    Find QChem files in a directory.
+
+    Only the mol.qout file (or alternatively files
+    with the task name as an extension, e.g., mol.qout.opt_0.gz, mol.qout.freq_1.gz, or something like this...)
+    will be returned.
+
+    Parameters
+    ----------
+    path
+        Path to a directory to search.
+
+    Returns
+    -------
+    Dict[str, Any]
+        The filenames of the calculation outputs for each QChem task, given as a ordered dictionary of::
+
+            {
+                task_name:{
+                    "qchem_out_file": qcrun_filename,
+                },
+                ...
+            }
+    If there is only 1 qout file task_name will be "standard" otherwise it will be the extension name like "opt_0"
```

Comment:
According to the docstring, this method is only supposed to return output files, but the regex pattern `in_file_pattern` will match _input_ files